core/tests/ts-integration/tests/interop.test.ts

```diff
@@ -23,580 +17,607 @@ import {
-// import { waitForBlockToBeFinalizedOnL1, waitUntilBlockFinalized } from '../src/helpers';
 
-export function readContract(path: string, fileName: string, contractName?: string) {
-    contractName = contractName || fileName;
-    return JSON.parse(fs.readFileSync(`${path}/${fileName}.sol/${contractName}.json`, { encoding: 'utf-8' }));
+// Read contract ABIs
+function readContract(path: string, fileName: string, contractName?: string) {
+	contractName = contractName || fileName;
+	return JSON.parse(fs.readFileSync(`${path}/${fileName}.sol/${contractName}.json`, { encoding: 'utf-8' }));
 }
-
-const bridgehubInterface = readContract(
-    '../../../contracts/l1-contracts/artifacts/contracts/bridgehub',
-    'Bridgehub'
-).abi;
-
-
-const ntvInterface = readContract(
-    '../../../contracts/l1-contracts/artifacts/contracts/bridge/ntv',
-    'L2NativeTokenVault'
-).abi;
-
-const l1Erc20ABI = [
-    'function mint(address to, uint256 amount)',
-    'function approve(address spender, uint256 value)',
-    'function balanceOf(address account) view returns (uint256)',
-    'function allowance(address owner, address spender) view returns (uint256)'
-];
-
+const ArtifactBridgeHub = readContract(
+	'../../../contracts/l1-contracts/artifacts/contracts/bridgehub',
+	'Bridgehub'
+);
+const ArtifactNativeTokenVault = readContract(
+	'../../../contracts/l1-contracts/artifacts/contracts/bridge/ntv',
+	'L2NativeTokenVault'
+);
+const ArtifactMintableERC20 = readContract(
+	'../../../contracts/l1-contracts/artifacts-zk/contracts/dev-contracts',
+	'TestnetERC20Token'
+);
+const ArtifactSwap = readContract('./artifacts-zk/contracts/Swap', 'Swap');
+
+// Constants
 const INTEROP_TX_TYPE = 253;
-const INTEROP_TX_TYPE_BIG_INT = 253n;
 
-describe('Interop checks', () => {
+describe("Interop checks", () => {
     let testMaster: TestMaster;
-    let alice: zksync.Wallet;
-    let bob: zksync.Wallet;
-    let aliceOtherChain: zksync.Wallet;
-    let bobOtherChain: zksync.Wallet;
-    let richWallet: zksync.Wallet;
-    let l2ProviderOtherChain: RetryProvider;
-    let l2Wallet: Wallet;
-    let sendingChainId = 271;
-    // let secondChainId = 272;
-    let secondChainId = 505;
-    let l1Bridgehub: ethers.Contract;
-    let tokenDetails: Token;
-    let assetId: string;
-    let zkAssetId: string;
-    let bridgehub: ethers.Contract;
-    let l2NativeTokenVault: ethers.Contract;
-    let l2NativeTokenVaultOtherChain: ethers.Contract;
-    let tokenAddressOtherChain: string;
-    const timeout = 10000;
-
-    let tokenErc20OtherChain: zksync.Contract;
-    let zkErc20: zksync.Contract;
-    let swap: ethers.Contract;
-    const richPk = '0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110';
 
+    // L1 Variables
+		let l1_provider: zksync.Provider;
+    let l1_wallet: zksync.Wallet;
+    let tokenA_details: {
+      assetId: string;
+      addresses: {
+        interop1: string;
+        interop2: string;
+      };
+    };
+    let tokenB_details: {
```

Comment:
tokenDetails struct should be used

Reply:
export interface Token {
    name: string;
    symbol: string;
    decimals: bigint;
    l1Address: string;
    l2Address_1: string;
    l2Address_2:string;
   assetId: string;
}